### PR TITLE
Add nudge to do expression error

### PR DESCRIPTION
**Pull request by pdgonzalez872**
Closes https://github.com/elixir-lang/elixir/issues/11551

### CODE REVIEW

File: lib/elixir/src/elixir_tokenizer.erl [hunk 1]
```diff
@@ -1479,7 +1479,11 @@ invalid_do_error(Prefix) ->
   "    end\n\n"
   "or the equivalent construct:\n\n"
   "    if(some_condition?, do: :this, else: :that)\n\n"
-  "where \"some_condition?\" is the first argument and the second argument is a keyword list"}.
+  "where \"some_condition?\" is the first argument and the second argument is a keyword list.\n\n"
+  "** Note:\n\n"
```
Comment (whatyouhide) [inline lib/elixir/src/elixir_tokenizer.erl:1483]: I love this message, but would not use the `** Note:` here. I'd just write the rest of the message 🙃

**Review by whatyouhide — APPROVED** [review]

File: lib/elixir/src/elixir_tokenizer.erl (hunk 1 above)
Comment (josevalim) [inline lib/elixir/src/elixir_tokenizer.erl:1483]:
```suggestion
```

Comment (whatyouhide) [reply lib/elixir/src/elixir_tokenizer.erl:1483]: Yes, should've added a suggestion, thanks José 😄

File: lib/elixir/src/elixir_tokenizer.erl [hunk 2]
```diff
@@ -1479,7 +1479,11 @@ invalid_do_error(Prefix) ->
...
+  "You may see this error if you forget a trailing comma before the \"do\" in a \"do\" expression.\n"
+  "This is common when using a `with` statement or a `for` comprehension spanning multiple lines.\n"
+  "If this is the case, please verify it is contructed correctly."}.
```
Comment (josevalim) [inline lib/elixir/src/elixir_tokenizer.erl:1486]:
```suggestion
  "You may see this error if you forget a trailing comma before the \"do\" in a \"do\" expression"}.
```

**Comment by josevalim** [comment]
Thank you, I did some changes and simplified it a bit. :) We need to be careful with specific wording in messages because it can be additionally confusing in different scenarios than the ones we originally consider.

File: lib/elixir/src/elixir_tokenizer.erl (hunk 1 above)
Comment (pdgonzalez872) [reply lib/elixir/src/elixir_tokenizer.erl]: 👍 works for me!

**Comment by pdgonzalez872** [comment]
No worries! Thanks @josevalim and @whatyouhide !

❤️

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**